math/geometry/distancebetweenpoints.go

```diff
@@ -0,0 +1,32 @@
+// distancebetweenpoints.go
+// Find distance between two points
+// Details:
+// Given coordinates of 2 points on a cartesian plane, find the distance between them rounded up to nearest integer.
+// Link: https://practice.geeksforgeeks.org/problems/distance-between-2-points3200/1?page=1&category[]=geometric
+
+// author(s) [Chetan Patil](https://github.com/Chetan07j)
+
+// Package geometry contains geometric algorithms
+package geometry
+
+import (
+	"math"
+)
+
+// This function calculates the distance based on Cartesian Plane Distance Formula
+// Formula: √((x1-x2) * (x1-x2) + (y1-y2) * (y1-y2))
+// Takes x and y coordinates of two point
+// Returns distance in floating number
+func FindDistanceBetweenTwoPoints(x1 float64, y1 float64, x2 float64, y2 float64) float64 {
+	// Find difference between x-coordinates
+	xDiff := x2 - x1
+
+	// Find difference between y-coordinates
+	yDiff := y2 - y1
+
+	// Apply formula
+	distance := (xDiff * xDiff) + (yDiff * yDiff)
+
+	// Return square root
+	return math.Sqrt(distance)
+}
```

Comment:
This should be more general IMO, making this general is relatively straight forward.

I think the following should work

```suggestion
type Point []float64

var ErrDimMismatch = errors.New("mismatched dimensions")

func EuclideanDistance(p1 Point, p2 Point) (float64, error) {
	n := len(p1)
	if len(p2) != n {
		return -1, ErrDimMismatch
	}
	diff_squared := make([]float64, n)
	for i, x_i := range p1 {
		diff := math.Abs(x_i - p2[i])
		diff_squared[i] = diff * diff
	}
	var total float64 = 0
	for _, val := range diff_squared {
		total += val
	}
	return math.Sqrt(total), nil
}
```

Reply:
@tjgurwara99 Thanks for your suggestion.
I have updated the code as per your suggestion but with slight modifications, could you please have a look at it again?